account_tax_rounding_method/models/account_tax.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountTax(models.Model):
+    _inherit = "account.tax"
+
+    @api.model
+    def _get_tax_rounding_method(
+        self,
+        company,
+        partners=None,
+    ):
+        if partners and len(partners) == 1 and partners[0].tax_rounding_method:
+            return partners[0].tax_rounding_method
+        return company.tax_rounding_method
+
+    @api.model
+    def _round_tax_details_tax_amounts(self, base_lines, company, mode="mixed"):
+        tax_rounding_method = self.env.context.get("tax_rounding_method")
+        if tax_rounding_method:
+            company = company.with_context(tax_rounding_method=tax_rounding_method)
+            for base_line in base_lines:
+                if "currency_id" in base_line and base_line["currency_id"]:
+                    currency = base_line["currency_id"]
+                    if isinstance(currency, int):
+                        currency = self.env["res.currency"].browse(currency)
+                    base_line["currency_id"] = currency.with_context(
+                        tax_rounding_method=tax_rounding_method
+                    )
```

Comment:
I thought this would be enough. I don't think there is any case of `currency_id` being int, IIUC.
```suggestion
                currency = base_line.get("currency_id")
                if currency:
                    base_line["currency_id"] = currency.with_context(
                        tax_rounding_method=tax_rounding_method
                    )
```